### DISCUSSION TITLE
how to add a new plugins to nvchad?

### DISCUSSION THREAD

**Discussion by alohl669** [NvChad/NvChad#1186] [General]
Hi, I work sometimes writing docs using markdown and I'm very interested to add any plugin like [spellsitter](https://github.com/lewis6991/spellsitter.nvim) to speed up the original vim spellchecking function. The problem is, I'm really bumbling with vim plugins, and seeing the spellsitter docs using packer.vim, and seeing the nvchad's directory tree I'm feeling lost. Sorry, anyone can help me? I suppose that I could do it using the custom diretory maybe?

**Comment by budchirp** [comment]
https://nvchad.github.io/config/Custom%20config#add-plugins

**Reply by alohl669** [reply]
I create `~/.config/nvim/lua/custom/plugins/init.lua` with

```
return {

   ["lewis6991/spellsitter.nvim"] = {
      config = function()
         require("spellsitter").setup()
      end
   },
}
```

and using :PackerInstall nothing happens. Same using `~/.config/nvim/lua/custom/init.lua` instead of `~/.config/nvim/lua/custom/plugins/init.lua`

If I type :PackerStatus, the plugin does not appear :(

**Reply by siduck** [reply]
did you load that file in chadrc?

**Reply by alohl669** [reply]
I guess not, I did only that I said. I created the `~/.config/nvim/lua/custom/init.lua` whit this content.

```
return {

   ["lewis6991/spellsitter.nvim"] = {
      config = function()
         require("spellsitter").setup()
      end
   },
}
```

**Reply by LeonHeidelbach** [reply]
@alohl669 Please read the documentation carefully. You will have to require the custom plugins table in your `chadrc.lua`. Specifically in the `M.plugins.user` field like this:

```lua
M.plugins = {
  -- custom plugin install path
  user = require(“<YOUR_CUSTOM_PLUGINS_MODULE_PATH>”),
}
```

If you need more specific examples please refer to @siduck ‘s example dotfiles. Also you cannot use the `init.lua` file directly in the custom folder for your mappings. Simply create a different file like `plugins.lua` within the custom folder or create a new directory as you apparently already did.

**Reply by alohl669** [reply]
Yeah!, already included and I see how it has been installed. But the plugin still not listed using :PackerStatus, I suppose this is normal, right?

NEW. I restarted nvim and finally, the plugin is listed.

Thank you so much

**Reply by siduck** [reply]
@alohl669 do know that nvim's config wont reload on the fly , you have to re-open nvim to see the changes

**Reply by mattbdc** [reply]
![image](https://user-images.githubusercontent.com/83773047/202651629-f145b6d0-930f-4ce3-98b8-01473abd7208.png)

Is this chadrc (no extension) at the root of .config/nvim/ ?  chadrc.lua at the root ? or chadrc  at lua/custom? or chadrc.lua at lua/custom? 

The convention further along suggests it would be a file path? but this conventions seems 50/50 (see later)
![image](https://user-images.githubusercontent.com/83773047/202651734-16ddec45-4e2b-4854-aa00-d87e28f8d22d.png)

But perhaps not?

![image](https://user-images.githubusercontent.com/83773047/202651772-eab20168-7f83-47db-92f5-ea7bd75fc4be.png)

What path is this file?  no docstring here.. 

I do think it's important that GH issues isn't globbered with repeat getting-go issues (for things like lspconfig) but the docs are far from clear here unless you have plugin/customization experience before

**Reply by siduck** [reply]
@mattbdc read https://nvchad.com/quickstart/post-install

please read the docs in order or you wont understand anything unless you're an existing nvchad user

**Reply by tarikkavaz** [reply]
> @mattbdc read https://nvchad.com/quickstart/post-install
> 
> please read the docs in order or you wont understand anything unless you're an existing nvchad user

https://nvchad.com/docs/quickstart/post-install

**Reply by sananand007** [reply]
Coming back to this, where is `Packer` now (in the sense I do not find it anymore within `~/.config/nvim` ) , is it not used anymore and it is replaced by `NvChadUpdate` . ?

**Reply by siduck** [reply]
@sananand007 we use lazy.nvim now, please read nvchad docs!

**Reply by sananand007** [reply]
thnx @siduck . Yes I was able to after some serious reading, Just on hindsight its not very fast and clear for folks who are not from the `Lua` or vim background and and new users. 
regardless would like to thanks for NVChad, I really love it !! 


For anyone who needs to install `toggleterm` using nvchad 2.0 , I post it here as this is likely the most information I found about this process. 

- After the below change, I just did `:NvChadUpdate` , the plugin was installed and functional. 

```lua
  --within ~/.config/nvim/lua/custom/plugins.lua
  -- Install ToggleTerm plugin here
  {
    "akinsho/toggleterm.nvim",
    cmd = {"ToggleTerm"},
    config = function()
      require("toggleterm").setup {
        size = 20,
        open_mapping = [[<c-\>]],
        hide_numbers = true, -- hide the number column in toggleterm buffers
        shade_filetypes = {},
        shade_terminals = true,
        shading_factor = 2, -- the degree by which to darken to terminal colour, default: 1 for dark backgrounds, 3 for light
        start_in_insert = true,
        insert_mappings = true, -- whether or not the open mapping applies in insert mode
        persist_size = false,
        direction = "float",
        close_on_exit = true, -- close the terminal window when the process exits
        shell = nil, -- change the default shell
        float_opts = {
        border = "rounded",
        winblend = 0,
        highlights = {
        border = "Normal",
        background = "Normal",
        },
      },
        winbar = {
          enabled = true,
          name_formatter = function(term) --  term: Terminal
            return term.count
          end,
        },
      }
      vim.cmd [[
      augroup terminal_setup | au!
      autocmd TermOpen * nnoremap <buffer><LeftRelease> <LeftRelease>i
      autocmd TermEnter * startinsert!
      augroup end
      ]]

      vim.api.nvim_create_autocmd({ "TermEnter" }, {
        pattern = { "*" },
        callback = function()
          vim.cmd "startinsert"
        end,
      })

      local opts = { noremap = true, silent = true }
      function _G.set_terminal_keymaps()
        vim.api.nvim_buf_set_keymap(0, "t", "<m-h>", [[<C-\><C-n><C-W>h]], opts)
        vim.api.nvim_buf_set_keymap(0, "t", "<m-j>", [[<C-\><C-n><C-W>j]], opts)
        vim.api.nvim_buf_set_keymap(0, "t", "<m-k>", [[<C-\><C-n><C-W>k]], opts)
        vim.api.nvim_buf_set_keymap(0, "t", "<m-l>", [[<C-\><C-n><C-W>l]], opts)
      end
    end
  }
```

**Reply by siduck** [reply]
@sananand007 soon from v3.0 you wouldnt need toggleterm, a new custom term module will replace nvterm

**Comment by wwwehr** [comment]
New link: https://nvchad.com/docs/config/plugins

**Reply by Ranguna** [reply]
Where do I put the `plugins.lua` file ?

I've used lazy before, but not with a plugins file.

**Reply by siduck** [reply]
@Ranguna check the starter repo, it has an example

**Reply by Ranguna** [reply]
@siduck this?
https://github.com/NvChad/starter/blob/main/lua%2Fplugins%2Finit.lua

**Reply by siduck** [reply]
@Ranguna  yes

**Reply by Ranguna** [reply]
Then should the docs be changed to show `lua/plugins/init.lua` instead of `plugins.lua` ?

**Reply by siduck** [reply]
@Ranguna  file name shouldnt matter if you're importing it in your main init.lua properly. Starter repo is just an example

**Reply by Ranguna** [reply]
@siduck they matter for beginners, and the starter repo is not just an example, it's _the_ example and it's mentioned in the docs already: https://nvchad.com/docs/quickstart/install

Maybe just a note referring to the starter example file path ?

**Reply by siduck** [reply]
@Ranguna honestly they dont matter even for the beginners. 

Any file under plugins dir returning a table will count as a plugin ( as per lazy.nvim's import feature ) . I renamed the plugins.lua in docs to `plugins/anyname.lua` . Check his config for example https://github.com/wSedlacek/nvim/tree/main/lua/plugins

**Reply by Ranguna** [reply]
Ah nice, that's even better. Thanks!

**Comment by daved30** [comment]
After 2.5, i simply cant add a plugin TT as the directory itself is marked as a red CROSS with the custom one atleast i was able to do so but now simply confused, i am simply enthusiastic over using nvchad but the complexity, maybe im just not made for this stuff, have been going through the docs for over a month now... at this point i just want a discord serv to connect with my fellow programmers to sort this issue out.

**Reply by siduck** [reply]
@daved30  join our server

**Reply by daved30** [reply]
joined the server thanx buddy now lets dive into nvChad!

**Reply by siduck** [reply]
@daved30 that one's just bridged to telegram, join the discord one..
>plugin TT as the directory itself is marked as a red CROSS with the custom

that just means that its not comitted with git yet. Please read the migration guide properly

example config https://github.com/wSedlacek/nvim/tree/main/lua/plugins

**Reply by daved30** [reply]
i cant see any other discord server links or buttons to join any other servers just could see one discord server with just 360 online... sorry, for the inconvenience.

**Reply by siduck** [reply]
@daved30  https://discord.com/invite/gADmkJb9Fb

**Comment by NicTanghe** [comment]


**Reply by siduck** [reply]
@NicTanghe , go to any file in your plugins/ dir and just add the plugin's table, pls read https://nvchad.com/docs/config/plugins

**Reply by NicTanghe** [reply]
I was bind.
